### PR TITLE
WRO-4903: Move moonstone related modules to moonstone repo.

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Moved all the moonstone-related only modules from enact/ui into moonstone/internal.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-4903.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ before_install:
     - sudo apt-get update
     - sudo apt-get install build-essential libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev
 install:
+    - npm cache clean --force
```
Comment (daniel-stoian-lgp) [inline .travis.yml]: remove this on the next commit

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The following is a curated list of changes in the Enact moonstone module, newest
 
 ### Added
 
+- `moonstone/internal/IconButton` component, transferred from `ui/IconButton`
+- `moonstone/internal/RadioDecorator` component, transferred from `ui/RadioDecorator`
+- `moonstone/internal/Scrollable` component, transferred from `ui/Scrollable`
+- `moonstone/internal/Scroller.UiScroller` component, transferred from `ui/Scroller.UiScroller`
+- `moonstone/internal/SlotItem` component, transferred from `ui/SlotItem`
+- `moonstone/internal/ToggleItem` component, transferred from `ui/ToggleItem`
+- `moonstone/internal/VirtualList.UiVirtalListBase` component, transferred from `ui/VirtualList.UiVirtalListBase`
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md]: I'm not sure what's the proper text for these additions. I wi;; let the next reviewer to handle it

File: ExpandableItem/Expandable.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import Cancelable from '@enact/ui/Cancelable';
 import hoc from '@enact/core/hoc';
-import RadioDecorator from '@enact/ui/RadioDecorator';
 import Toggleable from '@enact/ui/Toggleable';
 
 import ExpandableSpotlightDecorator from './ExpandableSpotlightDecorator';
+import RadioDecorator from '../internal/RadioDecorator';
```
Comment (daniel-stoian-lgp) [inline ExpandableItem/Expandable.js]: this import it's an internal import. it should be placed in a separate section, above ExpandableSpotlightDecorator

Please do it for all files, where applicable

File: CHANGELOG.md (hunk 2 above)
Comment (daniel-stoian-lgp) [inline CHANGELOG.md]:
```suggestion
- `moonstone/internal/VirtualList.UiVirtualListBase` component, transferred from `ui/VirtualList.UiVirtualListBase`
```

File: Scrollable/Scrollbar.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import ApiDecorator from '@enact/core/internal/ApiDecorator';
-import {ScrollbarBase as UiScrollbarBase} from '@enact/ui/Scrollable/Scrollbar';
+import {ScrollbarBase as UiScrollbarBase} from '../internal/Scrollable/Scrollbar';
```
Comment (daniel-stoian-lgp) [inline Scrollable/Scrollbar.js]: move this import to a different section

File: ToggleItem/ToggleItem.js [hunk 5]
```diff
@@ -26,11 +26,11 @@ import ForwardRef from '@enact/ui/ForwardRef';
 import Pure from '@enact/ui/internal/Pure';
 import PropTypes from 'prop-types';
 import Toggleable from '@enact/ui/Toggleable';
-import {ToggleItemBase as UiToggleItem} from '@enact/ui/ToggleItem';
 import Touchable from '@enact/ui/Touchable';
 import Spottable from '@enact/spotlight/Spottable';
 import compose from 'ramda/src/compose';
 
+import {ToggleItemBase as UiToggleItem} from '../internal/ToggleItem';
```
Comment (daniel-stoian-lgp) [inline ToggleItem/ToggleItem.js]: please update the jsdoc in this file  line138

File: VirtualList/VirtualListBase.js [hunk 6]
```diff
@@ -6,12 +6,12 @@ import Spotlight, {getDirection} from '@enact/spotlight';
 import Accelerator from '@enact/spotlight/Accelerator';
 import Pause from '@enact/spotlight/Pause';
 import {Spottable} from '@enact/spotlight/Spottable';
-import {VirtualListBase as UiVirtualListBase, VirtualListBaseNative as UiVirtualListBaseNative} from '@enact/ui/VirtualList';
 import PropTypes from 'prop-types';
 import clamp from 'ramda/src/clamp';
 import {Fragment, Component} from 'react';
 import warning from 'warning';
 
+import {VirtualListBase as UiVirtualListBase, VirtualListBaseNative as UiVirtualListBaseNative} from '../internal/VirtualList';
```
Comment (daniel-stoian-lgp) [inline VirtualList/VirtualListBase.js]: check line869 and 882

File: samples/sampler/stories/qa/VirtualGridList.js [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
 import Item from '@enact/moonstone/Item';
 import {VirtualGridList, VirtualListBase} from '@enact/moonstone/VirtualList';
 import {action} from '@enact/storybook-utils/addons/actions';
 import {boolean, number, select} from '@enact/storybook-utils/addons/knobs';
 import {mergeComponentMetadata} from '@enact/storybook-utils';
 import ri from '@enact/ui/resolution';
-import {ScrollableBase as UiScrollableBase} from '@enact/ui/Scrollable';
-import {VirtualListBase as UiVirtualListBase} from '@enact/ui/VirtualList/VirtualListBase';
+
+import {VirtualListBase as UiVirtualListBase} from '@enact/moonstone/internal/VirtualList';
```
Comment (daniel-stoian-lgp) [inline samples/sampler/stories/qa/VirtualGridList.js]: move this improt to proper position

File: samples/sampler/stories/qa/VirtualList.js [hunk 8]
```diff
@@ -9,8 +10,7 @@ import {action} from '@enact/storybook-utils/addons/actions';
 import {boolean, number, select} from '@enact/storybook-utils/addons/knobs';
 import {mergeComponentMetadata} from '@enact/storybook-utils';
 import ri from '@enact/ui/resolution';
-import {ScrollableBase as UiScrollableBase} from '@enact/ui/Scrollable';
-import {VirtualListBase as UiVirtualListBase} from '@enact/ui/VirtualList';
+import {VirtualListBase as UiVirtualListBase} from '@enact/moonstone/internal/VirtualList';
```
Comment (daniel-stoian-lgp) [inline samples/sampler/stories/qa/VirtualList.js]: same here

File: .travis.yml (hunk 1 above)
Comment (alexandrumorariu) [reply .travis.yml]: Done.

File: samples/sampler/stories/qa/VirtualGridList.js (hunk 7 above)
Comment (alexandrumorariu) [reply samples/sampler/stories/qa/VirtualGridList.js]: Done

File: samples/sampler/stories/qa/VirtualList.js (hunk 8 above)
Comment (alexandrumorariu) [reply samples/sampler/stories/qa/VirtualList.js]: Done

File: VirtualList/VirtualListBase.js (hunk 6 above)
Comment (alexandrumorariu) [reply VirtualList/VirtualListBase.js]: Done

File: ToggleItem/ToggleItem.js (hunk 5 above)
Comment (alexandrumorariu) [reply ToggleItem/ToggleItem.js]: Done

File: Scrollable/Scrollbar.js (hunk 4 above)
Comment (alexandrumorariu) [reply Scrollable/Scrollbar.js]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (alexandrumorariu) [reply CHANGELOG.md]: Done

File: ExpandableItem/Expandable.js (hunk 3 above)
Comment (alexandrumorariu) [reply ExpandableItem/Expandable.js]: Done

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Comment by MikyungKim** [comment]
@alexandrumorariu @daniel-stoian-lgp, would you please update the branch with the recent develop branch? Thank you.

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,6 +2,18 @@
 
 The following is a curated list of changes in the Enact moonstone module, newest changes on the top.
 
+## [unreleased]
+
+### Added
+
+- `moonstone/ui/IconButton` component, transferred from `ui/IconButton`
+- `moonstone/ui/RadioDecorator` component, transferred from `ui/RadioDecorator`
+- `moonstone/ui/Scrollable` component, transferred from `ui/Scrollable`
```
Comment (MikyungKim) [inline CHANGELOG.md]: Please remove private modules here.

File: ToggleItem/ToggleItem.js [hunk 10]
```diff
@@ -135,7 +135,7 @@ const defaultConfig = {
  *
  * @class ToggleItemDecorator
  * @memberof moonstone/ToggleItem
- * @mixes ui/ToggleItem.ToggleItemDecorator
+ * @mixes moonstone/ui/ToggleItem.ToggleItemDecorator
```
Comment (MikyungKim) [inline ToggleItem/ToggleItem.js]: Is this the right one? I don't see `ui/ToggleItemDecoroator`

File: ui/IconButton/IconButton.module.less [hunk 11]
```diff
@@ -0,0 +1,18 @@
+// IconButton.module.less
+//
+@import '../../styles/ui/mixins';
```
Comment (MikyungKim) [inline ui/IconButton/IconButton.module.less]: 🤔 I think we can refer to ui's mixins not from here redefined.

File: ui/RadioDecorator/RadioControllerDecorator.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * A higher-order component that establishes a radio group context for its descendants.
+ *
+ * Any descendants that are wrapped by {@link moonstone/ui/RadioDecorator.RadioDecorator} will be mutually exclusive.
+ *
+ * @class RadioControllerDecorator
+ * @memberof moonstone/ui/RadioDecorator
+ * @hoc
+ * @private
```
Comment (MikyungKim) [inline ui/RadioDecorator/RadioControllerDecorator.js]: If the ui's module was public, let's keep public here too. Please check all of the components added.

File: ToggleItem/ToggleItem.js (hunk 10 above)
Comment (daniel-stoian-lgp) [reply ToggleItem/ToggleItem.js]: Yes,. ToggleItemDecorator is in the ui/ToggleItem/ToggleItem.js file

File: ui/IconButton/IconButton.module.less (hunk 11 above)
Comment (daniel-stoian-lgp) [reply ui/IconButton/IconButton.module.less]: I totally agree. I didn't like this redefinition yesterday when I revisited the code. I was just waiting for a second opinion

File: ui/IconButton/IconButton.module.less [hunk 13]
```diff
@@ -0,0 +1,18 @@
+// IconButton.module.less
+//
+@import '@enact/ui/styles/mixins';
```
Comment (MikyungKim) [inline ui/IconButton/IconButton.module.less]:
```suggestion
@import '~@enact/ui/styles/mixins.less';
```

**Review by MikyungKim — APPROVED** [review]
LGTM

**Comment by seunghoh** [comment]
Nice work!

**Pull request merged**